### ISSUE TITLE
[Feature Request]: Audit Checkin/Checkout

### ISSUE THREAD

**Issue by ShaunNeighbourSwale** [grokability/snipe-it#15116]
### Is your feature request related to a problem? Please describe.

When auditing assets, if the location is incorrect, we can only update the default asset location from the audit screen - Not the checked out location.

### Describe the solution you'd like

The ability to update the checked out location on the audit screen.

**Comment by snipe** [comment]
Hi @ShaunNeighbourSwale - can you give me a use-case where this would be desired? The asset's checked out location generally follows the person/place/etc that it's checked out to.

**Comment by ShaunNeighbourSwale** [comment]
Hi @snipe 

Currently, when performing an audit, if an asset is found to have been moved, and it's not appropriate to move it back, the auditor would need to complete the audit, then separately find the asset again, check it in, and check it out to the new location.

Updating the checked out location during audit, would be considerably more useful than changing the asset's default location (which rightly/wrongly we use to indicate the location of an asset when it's not deployed).  The vast majority of our assets are deployed, so the majority of audits ignore the default location, and would be focused of the checked out location.

Regards

**Comment by snipe** [comment]
According to the code, it doesn't update the `rtd_location_id`, it updates the actual location:

https://github.com/snipe/snipe-it/blob/46caa478d2eb120f3a1acec3b04dc4c9866e6116/app/Http/Controllers/Assets/AssetsController.php#L912-L916

**Comment by ShaunNeighbourSwale** [comment]
![image](https://github.com/user-attachments/assets/ca95554f-cd1a-4e77-8a31-720b1f12b068)

**Comment by snipe** [comment]
That's correct. It does not change the location of the *target*, just of the asset itself.

**Comment by ShaunNeighbourSwale** [comment]
OK - So could you consider adding a box so that this could be updated during the audit process?

**Comment by snipe** [comment]
That's not what you were asking for though. The `location_id` on the asset is what's being updated here and what your original ask was for. Not all admins with permission to checkin/checkout assets will have the permission to change user's locations.

**Comment by ShaunNeighbourSwale** [comment]
Sorry for any confusion.

We're auditing assets - Not users.

We're not setting the asset location separately - This is set when an asset is checked out.

When performing an audit of an asset, you can indeed update the asset location - But the asset is still checked out to another location.......  That doesn't make any sense.

**Comment by snipe** [comment]
>  But the asset is still checked out to another location....... That doesn't make any sense.

That's unfortunately not a lot clearer. Are you checking things out to locations or to users?

**Comment by ShaunNeighbourSwale** [comment]
We checkout assets to either locations, other assets or users.

In this use case, we audit a room (location), find several assets that have been moved, so need to update their location as part of the audit.

For context, here's an example workflow;

1. Asset is created.  The company_id and rtd_location_id are specified during creation.
2. The asset is checked-out to a building/room - This process appears to set the location_id?
3. The asset is moved by an end user to a different room.
4. An audit is carried out - The asset is scanned, and found to be in the wrong room.  To update the asset, we now have to complete the audit process, then go back to the asset record to check it in, then go back to the asset record again and finally check it out to the new room.

**Comment by snipe** [comment]
That's not what that's for though. If the asset is assigned to a _**location**_, you have to check it in and back out to the new location. That's the point of tracking the history. Changing the location there in the audit screen would be for if the asset is still checked out to User XYZ, but it's now located in their work office instead of their home office. The responsibility still lies with that user and the asset is still checked out to them, but the physical location of where is exists has changed.

**Comment by ShaunNeighbourSwale** [comment]
OK - But that makes the current audit process pretty useless if you're not exclusively checking out assets to users.

In our instance, the vast majority of assets are checked out to locations.

**Comment by snipe** [comment]
We've always discouraged people from checking out things to non-users, as there is no responsible party, which sort of defeats the purpose of using an asset tracking system. When an asset goes missing from a location or another asset, who is responsible? Or if it gets damaged? 

Auditing is meant to make sure thing the still exists and that it's in good (or at least expected) working order, and in the place it was before. It's not really meant to be a checkin/checkout process. Not everybody uses audits, but everybody uses the checkin/checkout utilities - just depends on org requirements. 

I'll have to see if I can find a way to work within your use case. In the meantime, you might want to change your workflow to use expected checkin date as the "audit", since it will be easier to check the asset in from the one location and out to another.

**Comment by ShaunNeighbourSwale** [comment]
Thanks for considering the feature request.

Regarding checkouts to non-users.  There's not really an alternative here;

- We have multiple teachers that could be responsible for the assets in a classroom throughout the day.  They are expected to report issues, which would indicate that any damage occurred during the previous lesson.  But it's not practical to check-out 60 assets to each teacher for each classroom they're currently teaching in.
- We have to track both deployed and ready-to-deploy assets.  If we simply change an asset location, that doesn't indicate that it has been deployed - Only checking out the asset does this.  We need to be able to quickly and easily identify the difference between a desktop computer that's in use on a desk, to one that's sat on a spares shelf above that desk.  Deployed vs Ready to Deploy.
- We assign laptops to laptop trolleys, because it's a damned sight quicker to checkout a trolley to a new location than it is to checkin/checkout 30 laptops.

While I'm sure there is a specific intended way to run Snipe-IT,  that may not suit us.  It is an excellent product, and the support is well above anything else we've used.  But I must have missed the memo regarding how I was supposed to use it, and now we're 20,000+ assets deep, and need to do regular audits to figure out where teachers have hidden all our Chromebooks........

Thanks again.
